flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveTableUtil.java

```diff
@@ -71,4 +80,37 @@ public static TableSchema createTableSchema(List<FieldSchema> cols, List<FieldSc
 
 		return columns;
 	}
+
+	// --------------------------------------------------------------------------------------------
+	//  Helper methods
+	// --------------------------------------------------------------------------------------------
+
+	/**
+	 * Create properties info to initialize a SerDe.
+	 * @param storageDescriptor
+	 * @return
+	 */
+	public static Properties createPropertiesFromStorageDescriptor(StorageDescriptor storageDescriptor) {
+		SerDeInfo serDeInfo = storageDescriptor.getSerdeInfo();
+		Map<String, String> parameters = serDeInfo.getParameters();
+		Properties properties = new Properties();
+		properties.setProperty(
+				serdeConstants.SERIALIZATION_FORMAT,
+				parameters.get(serdeConstants.SERIALIZATION_FORMAT));
+		List<String> colTypes = new ArrayList<>();
+		List<String> colNames = new ArrayList<>();
+		List<FieldSchema> cols = storageDescriptor.getCols();
+		for (FieldSchema col: cols){
+			colTypes.add(col.getType());
+			colNames.add(col.getName());
+		}
+		properties.setProperty(serdeConstants.LIST_COLUMNS, StringUtils.join(colNames, ","));
+		// Note: serdeConstants.COLUMN_NAME_DELIMITER is not defined in previous Hive. We use a literal to save on shim
+		properties.setProperty("column.name.delimite", String.valueOf(SerDeUtils.COMMA));
+		properties.setProperty(serdeConstants.LIST_COLUMN_TYPES, StringUtils.join(colTypes, DEFAULT_LIST_COLUMN_TYPES_SEPARATOR));
```

Comment:
why join column names with `,` in line 107 but join column types with `:` here? Can we just use comma to be consistent?

Reply:
in line 107, it is a hive inner design. Hive use ':'/','/';' as columnType delimiter.